### ISSUE TITLE
Issues with replaceTrack, will statsended fire or give me what I want.

### ISSUE THREAD

**Issue by vr000m** [w3c/webrtc-stats#472]
With all the changes to track attachment stats, we are close to deprecating it. May be we should not. 

My use-case of replaceTrack switching from camera to screenshare, this does not fire an event on the peerconnection. We kind of solved this issue with `onstatsended` event returning the stats of track that was being replaced. however, with the recent changes, if we get rid of the empty dictionaries there will be no stats for the track that got replaced, that means the onstatsended will fire but return an empty dictionary. We would then need to call getStats() and get the latest stats from there, albeit with a bit of delay.

**Comment by henbos** [comment]
Some possible ways forward:
1. Don't have an event that fires because of replaceTrack; we can probably remove "onstatsended" completely.
2. Have an event that fires due to replaceTrack, but make it clear that this is what it really does, since for your use case you would probably want to invoke getStats() anyway - what is of interest is not the ended stats per-se but the fact that a significant event occurred at this point in time. Replace "onstatsended" with "onreplacetrack".
3. Make outbound-rtp end when attachment changes, so that outbound-rtp can be used the same way that "track" was intended.

**Comment by alvestrand** [comment]
The reason for having "onstatsended" was to be able to put in specifications for objects where you couldn't get stats for them after they were gone. I think address pair stats after an ICE restart was the other classic case. So #1 is not of great benefit.

In general mediating "something significant happened" through stats indicates that we're lacking this capability elsewhere in the model, and we should solve this there, not here.

**Comment by henbos** [comment]
Oops, I filed a duplicate issue here #480

**Comment by henbos** [comment]
TPAC decision: Remove "onstatsended" and file a separate issue for "onstatsevent" or "onreplacetrack" or what else to do.

**Comment by henbos** [comment]
The TPAC decision means we can close this after #497 and corresponding webrtc-pc PR has been merged, but we need to file an issue on webrtc-pc or webrtc-nv-use-cases?

**Comment by henbos** [comment]
onstatsended have now been removed from webrtc-pc: https://github.com/w3c/webrtc-pc/issues/2305

**Comment by henbos** [comment]
Should we close this or do you want to keep it open @vr000m? If you want to propose alternative events I think you should go to webrtc-nv-use-cases
